File: client/lib/core/models/manga_metadata.dart
class MangaMetadata {
  final String mediaId;
  final String? title;
  final String? synopsis;
  final String? genres;
  final double? score;
  final String? status;
  final int? year;
  final int? malId;
  final int? anilistId;
  final int? comicvineId;
  final String? source;
  final DateTime? lastScrapedAt;

  const MangaMetadata({
    required this.mediaId,
    this.title,
    this.synopsis,
    this.genres,
    this.score,
    this.status,
    this.year,
    this.malId,
    this.anilistId,
    this.comicvineId,
    this.source,
    this.lastScrapedAt,
  });

  factory MangaMetadata.fromJson(Map<String, dynamic> j) => MangaMetadata(
        mediaId:     j['mediaId'] as String,
        title:       j['title'] as String?,
        synopsis:    j['synopsis'] as String?,
        genres:      j['genres'] as String?,
        score:       (j['score'] as num?)?.toDouble(),
        status:      j['status'] as String?,
        year:        j['year'] as int?,
        malId:       j['malId'] as int?,
        anilistId:   j['anilistId'] as int?,
        comicvineId: j['comicvineId'] as int?,
        source:      j['source'] as String?,
        lastScrapedAt: j['lastScrapedAt'] != null
            ? DateTime.tryParse(j['lastScrapedAt'] as String)
            : null,
      );

  List<String> get genreList =>
      genres?.split(',').map((g) => g.trim()).where((g) => g.isNotEmpty).toList() ?? [];
}

File: server/Rekindle.Core/Models/MangaMetadata.cs
namespace Rekindle.Core.Models;

public class MangaMetadata
{
    public string MediaId { get; set; } = string.Empty;
    public string? Title { get; set; }
    public string? Synopsis { get; set; }
    /// <summary>Comma-separated list of genres.</summary>
    public string? Genres { get; set; }
    public double? Score { get; set; }
    public string? Status { get; set; }
    public int? Year { get; set; }
    public int? MalId { get; set; }
    public int? AnilistId { get; set; }
    public int? ComicvineId { get; set; }
    /// <summary>"mal" | "anilist" | "comicvine"</summary>
    public string? Source { get; set; }
    public DateTime? LastScrapedAt { get; set; }
}
